### ISSUE TITLE
ios_ntp_global always renders prefer on ntp server/peer

### ISSUE THREAD

**Issue by ledgley** [ansible-collections/cisco.ios#434]
##### SUMMARY
ios_ntp_global is always rendering `prefer` for ntp server and ntp peer statements

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ios_ntp_global

##### ANSIBLE VERSION
```
ansible 2.9.22
  config file = /Users/edgleyl/.ansible.cfg
  configured module search path = ['/Users/edgleyl/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/edgleyl/venv/two-nine/lib/python3.7/site-packages/ansible
  executable location = /Users/edgleyl/venv/two-nine/bin/ansible
  python version = 3.7.7 (default, Mar 10 2020, 15:43:03) [Clang 11.0.0 (clang-1100.0.33.17)]
```
and
```
ansible 2.10.12
  config file = /Users/edgleyl/.ansible.cfg
  configured module search path = ['/Users/edgleyl/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/edgleyl/venv/two-ten/lib/python3.7/site-packages/ansible
  executable location = /Users/edgleyl/venv/two-ten/bin/ansible
  python version = 3.7.7 (default, Mar 10 2020, 15:43:03) [Clang 11.0.0 (clang-1100.0.33.17)]
```

##### COLLECTION VERSION
```
# /Users/edgleyl/venv/two-ten/lib/python3.7/site-packages/ansible_collections
Collection Version
---------- -------
cisco.ios  2.5.0

# /Users/edgleyl/.ansible/collections/ansible_collections
Collection Version
---------- -------
cisco.ios  2.5.0
```

##### CONFIGURATION
```
(two-ten) edgleyl@edgleyl-JSS1242 compliance % ansible-config dump --only-changed
DEFAULT_CALLBACK_WHITELIST(/Users/edgleyl/.ansible.cfg) = ['timer']
HOST_KEY_CHECKING(/Users/edgleyl/.ansible.cfg) = False
PERSISTENT_COMMAND_TIMEOUT(/Users/edgleyl/.ansible.cfg) = 180
```

##### OS / ENVIRONMENT
MacOS 11.6
Using `state: rendered` so I don't think device firmware is relevant.

##### STEPS TO REPRODUCE
Use ios_ntp_global module to render device specific configuration

```yaml
  tasks:

    - name: CONFIGURE NTP
      cisco.ios.ios_ntp_global:
        config:
          peers:
            - peer: 204.20.240.1
              prefer: True
            - peer: 204.20.240.2
              prefer: False
          servers:
            - server: 172.24.99.229
              prefer: True
            - server: 172.23.99.229
              prefer: False
        state: rendered
      register: output

    - debug: var=output.rendered

    - debug: var=output.rendered
```

##### EXPECTED RESULTS
```
TASK [debug] ***********************************************************************************************************************************************
ok: [mw-prii-s01] => {
    "output.rendered": [
        "ntp peer 204.20.240.1 prefer",
        "ntp peer 204.20.240.2",
        "ntp server 172.24.99.229 prefer",
        "ntp server 172.23.99.229"
    ]
}
```

##### ACTUAL RESULTS
Though only one NTP peer and one NTP Server list item are set to `prefer: True` all are rendered with prefer.
```
TASK [debug] ***********************************************************************************************************************************************
ok: [mw-prii-s01] => {
    "output.rendered": [
        "ntp peer 204.20.240.1 prefer",
        "ntp peer 204.20.240.2 prefer",
        "ntp server 172.24.99.229 prefer",
        "ntp server 172.23.99.229 prefer"
    ]
}
```

**Comment by KB-perByte** [comment]
Hey @ledgley, The attributes are considered in a `false` state when the behavior is defined for a false value, it is not recommended to specify an attribute unless it is required to process. And as the false state for `prefer` is not there for peer or server.  You are supposed to get the expected output on the below example.

```
  tasks:
    - name: CONFIGURE NTP
      cisco.ios.ios_ntp_global:
        config:
          peers:
            - peer: 204.20.240.1
              prefer: True
            - peer: 204.20.240.2
          servers:
            - server: 172.24.99.229
              prefer: True
            - server: 172.23.99.229
        state: rendered
      register: output

    - debug: var=output.rendered

    - debug: var=output.rendered
```

**Comment by ledgley** [comment]
Thank you @KB-perByte I understand your logic however, per the documentation the module accepts a boolean value for this parameter.

Furthermore, rather than processing the False value, the module is actually inverting the boolean value given. If I define `prefer: False` the module is treating the fact that I defined the parameter as if I defined `prefer: True`. It is ignoring the False value.

It's fine to say that convention says not to explicitly define non-required parameters, though this module is dishonoring the input given, resulting in the opposite of what is required.
